File: app/src/main/java/uk/co/transferx/app/tutorial/adapter/TutorialAdapter.java
package uk.co.transferx.app.tutorial.adapter;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import uk.co.transferx.app.R;
import uk.co.transferx.app.tutorial.fragments.TutorialFragment;

import static uk.co.transferx.app.util.Constants.BUTTON_TEXT;
import static uk.co.transferx.app.util.Constants.DESCRIPTION_ONE;
import static uk.co.transferx.app.util.Constants.DESCRIPTION_TWO;
import static uk.co.transferx.app.util.Constants.LAYOUT;
import static uk.co.transferx.app.util.Constants.TUTORIAL_IMG;

/**
 * Created by sergey on 21/03/2018.
 */

public class TutorialAdapter extends FragmentStatePagerAdapter {

    private final static int PAGES = 5;
    private final static int FIRST_PAGE = 0;
    private final static int SECOND_PAGE = 1;
    private final static int THIRD_PAGE = 2;
    private final static int FORTH_PAGE = 3;
    private final static int FIVE_PAGE = 4;


    public TutorialAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        Bundle bundle = new Bundle();
        switch (position) {
            case FIRST_PAGE:
                bundle.putInt(LAYOUT, R.layout.tutorial_fragment_first);
                TutorialFragment firstTutorialPage = new TutorialFragment();
                firstTutorialPage.setArguments(bundle);
                return firstTutorialPage;
            case SECOND_PAGE:
                bundle.putInt(LAYOUT, R.layout.tutorial_fragment_layout);
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_security);
                bundle.putInt(DESCRIPTION_ONE, R.string.security);
                bundle.putInt(DESCRIPTION_TWO, R.string.security_desc);
                TutorialFragment secondTutorialPage = new TutorialFragment();
                secondTutorialPage.setArguments(bundle);
                return secondTutorialPage;
            case THIRD_PAGE:
                bundle.putInt(LAYOUT, R.layout.tutorial_fragment_layout);
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_auto_transfers);
                bundle.putInt(DESCRIPTION_ONE, R.string.automated_transfer);
                bundle.putInt(DESCRIPTION_TWO, R.string.second_page_desc);
                TutorialFragment threeTutorialPage = new TutorialFragment();
                threeTutorialPage.setArguments(bundle);
                return threeTutorialPage;
            case FORTH_PAGE:
                bundle.putInt(LAYOUT, R.layout.tutorial_fragment_layout_last);
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_recipients);
                bundle.putInt(DESCRIPTION_ONE, R.string.recipients);
                bundle.putInt(DESCRIPTION_TWO, R.string.three_page_desc);
                bundle.putInt(BUTTON_TEXT, R.string.add_recipientss);
                TutorialFragment fourTutorialPage = new TutorialFragment();
                fourTutorialPage.setArguments(bundle);
                return fourTutorialPage;
            case FIVE_PAGE:
                bundle.putInt(LAYOUT, R.layout.tutorial_fragment_layout_last);
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_credit_card);
                bundle.putInt(DESCRIPTION_ONE, R.string.title_flex);
                bundle.putInt(DESCRIPTION_TWO, R.string.flexible_payment_desc);
                bundle.putInt(BUTTON_TEXT, R.string.add_payment_method);
                TutorialFragment fiveTutorialPage = new TutorialFragment();
                fiveTutorialPage.setArguments(bundle);
                return fiveTutorialPage;
            default:
                throw new IllegalStateException("item is illegal" + position);
        }
    }

    @Override
    public int getCount() {
        return PAGES;
    }
}

File: app/src/main/java/uk/co/transferx/app/tutorial/adapter/TutorialAdapter.kt
package uk.co.transferx.app.tutorial.adapter

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.content.ContextCompat
import uk.co.transferx.app.R
import uk.co.transferx.app.tutorial.fragments.TutorialFragment
import uk.co.transferx.app.util.Constants.*

/**
 * Created by sergey on 21/03/2018.
 */

class TutorialAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        val bundle = Bundle()
        bundle.putInt(BUTTON_TEXT, R.string.skip)

        when (position) {
            FIRST_PAGE -> {
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_security)
                bundle.putInt(TITLE, R.string.security)
                bundle.putInt(DESCRIPTION, R.string.security_desc)
                bundle.putInt(BACKGROUND_COLOR, R.color.light_blue_tutorial)
            }
            SECOND_PAGE -> {
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_auto_transfers)
                bundle.putInt(TITLE, R.string.automated_transfer)
                bundle.putInt(DESCRIPTION, R.string.second_page_desc)
                bundle.putInt(BACKGROUND_COLOR, R.color.dark_blue_tutorial)
            }
            THIRD_PAGE -> {
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_recipients)
                bundle.putInt(TITLE, R.string.recipients)
                bundle.putInt(DESCRIPTION, R.string.three_page_desc)
                bundle.putInt(BACKGROUND_COLOR, R.color.black_tutorial)
            }
            FORTH_PAGE -> {
                bundle.putInt(TUTORIAL_IMG, R.drawable.ic_credit_card)
                bundle.putInt(TITLE, R.string.title_flex)
                bundle.putInt(DESCRIPTION, R.string.flexible_payment_desc)
                bundle.putInt(BACKGROUND_COLOR, R.color.purple_tutorial)
                bundle.putInt(BUTTON_TEXT, R.string.get_started)
            }
            else -> throw IllegalStateException("item is illegal$position")
        }

        val currentTutorialFragment = TutorialFragment()
        currentTutorialFragment.arguments = bundle

        return currentTutorialFragment
    }

    override fun getCount(): Int {
        return PAGES
    }

    companion object {
        private const val PAGES = 4
        private const val FIRST_PAGE = 0
        private const val SECOND_PAGE = 1
        private const val THIRD_PAGE = 2
        private const val FORTH_PAGE = 3
    }
}
